lib/mutant/mutator/node/regexp.rb

```diff
@@ -0,0 +1,104 @@
+module Mutant
+  class Mutator
+    class Node
+
+      # Generic mutator
+      module RegularExpression
+        class Parser
+          def self.parse(source)
+            Node.new(Regexp::Parser.parse(source, 'ruby/2.3'))
+          end
+
+          class Node < DelegateClass(Regexp::Expression::Base)
```

Comment:
I think writing a bijective mapper from `Regexp::Expression` to `Parser::AST` is not that complex and will be worth it so we can keep the mutable API contained. It'll not leave the mutation engine for regexps.